# Mon deuxième carnet : jouons avec les données sur les demandes
# d'accès à l'information au niveau fédéral

In [1]:
import pandas as pan

In [2]:
info = pan.read_csv("ati.csv")
#Ici, j'ai demandé au carnet de lire mon fichier csv, que j'ai placé
#Dans le même fichier que mon nouveau carnet.

In [3]:
info

year  month request_number  \
0      2016      3   A-2015-00070   
1      2017      5   A-2016-00059   
2      2016      3   A-2015-00059   
3      2016      7   A-2016-00076   
4      2016      3   A-2015-00113   
5      2016      3   A-2015-00109   
6      2016      3   A-2015-00108   
7      2016      3   A-2015-00107   
8      2016      3   A-2015-00082   
9      2016      3   A-2015-00106   
10     2016      3   A-2015-00100   
11     2016      3   A-2015-00098   
12     2016      3   A-2015-00085   
13     2016      3   A-2015-00084   
14     2016      3   A-2015-00063   
15     2016      3   A-2015-00061   
16     2016      3   A-2015-00058   
17     2016      3   A-2015-00057   
18     2016      3   A-2015-00052   
19     2016      3   A-2014-00183   
20     2016      3   A-2014-00182   
21     2016      3   A-2014-00161   
22     2016      3   A-2014-00160   
23     2016      7   A-2016-00071   
24     2017      7   A-2017-00063   
25     2016     10   A-2016-00121   
26     2016      4    A-2016-0002   
27     2016      4   A-2015-00112   
28     2016      4   A-2015-00103   
29     2016      4   A-2015-00101   
...     ...    ...            ...   
28264  2017      5        2017-01   
28265  2016     10        2016-07   
28266  2016     11        2016-20   
28267  2017      2        2016-22   
28268  2017      4        2016-21   
28269  2017      5        2016-12   
28270  2017      5        2016-14   
28271  2017      5        2016-15   
28272  2017     11        2016-05   
28273  2017     11        2016-13   
28274  2017      3   A-2016-00009   
28275  2016     10   A-2016-00003   
28276  2016     10   A-2016-00006   
28277  2016     10   A-2016-00007   
28278  2016     11   A-2016-00004   
28279  2016      8   A-2016-00002   
28280  2016      6   A-2016-00001   
28281  2017      2   A-2016-00011   
28282  2017      2   A-2016-00013   
28283  2017      3   A-2016-00014   
28284  2017      1   A-2016-00010   
28285  2017      4   A-2016-00012   
28286  2017      4   A-2016-00015   
28287  2017      5   A-2016-00016   
28288  2017      5   A-2016-00017   
28289  2017      7   A-2017-00002   
28290  2017      7   A-2017-00003   
28291  2017      7   A-2017-00004   
28292  2017     10   A-2017-00005   
28293  2017     12   A-2017-00009   

                                              summary_en  \
0      Provide a copy of the Transitional Briefing Bo...   
1      All communications between the AgriMarketing P...   
2      Copies  of any and all briefing materials prep...   
3      The amount of funding provided to National Far...   
4      All  records relating to a 2015-2016 outbreak ...   
5      Any communications, since January, 2015, betwe...   
6      Forensic audit report as described in Merx Sol...   
7      Pin-to-Pin messages between Pierre Corriveau a...   
8             Policy briefing book for the new minister    
9      Any communications on or instructions given to...   
10      A list of briefing materials, notes or memos ...   
11     All information about which company was funded...   
12     Copies of any briefing books or transition boo...   
13     A copy of the briefing book/binder/notes prepa...   
14     All briefing notes, memos, talking points, min...   
15     Provide access to last available draft or copy...   
16     Copies of any briefing books or transition boo...   
17     I would like to receive a copy of all briefing...   
18     Records created by AAFC Programs Branch (Farm ...   
19     Table 1: Projects Conducted With Neonicotinoid...   
20     Table 1: Projects Conducted With Neonicotinoid...   
21     Table 1: Projects Conducted With Neonicotinoid...   
22     Table 1: Projects Conducted With Neonicotinoid...   
23     The amount of funding provided to National Far...   
24     Copy of the briefing note (#226947) UPDATE ON ...   
25     Please provide copies of the tobacco farming d...   
26     Farm-level micro data solely for dairy, poultr...   
27     Any damage assessments, e

## C'est à partir d'ici que nous commencerons à manipuler et nettoyer les informations pour essayer d'aller retirer des choses intéressantes

In [4]:
#La première chose que je fais faire, c'est de nettoyer le nom des colonnes
#de mon tableau pour le rendre plus facile à travailler.

info.columns
#J'utilise cette commande qui va m'afficher les éléments de la première
#ligne sous forme de liste. Je vais ensuite la copier et changer les
#éléments les un après les autres.

Index(['year', 'month', 'request_number', 'summary_en', 'summary_fr',
       'disposition', 'pages', 'owner_org', 'owner_org_title'],
      dtype='object')

In [5]:
info.columns = ['annee', 'mois', 'n_de_requete', 'resume_an', 'resume_fr',
       'disposition', 'pages', 'owner_org', 'proprietaire']

In [6]:
info

annee  mois  n_de_requete  \
0       2016     3  A-2015-00070   
1       2017     5  A-2016-00059   
2       2016     3  A-2015-00059   
3       2016     7  A-2016-00076   
4       2016     3  A-2015-00113   
5       2016     3  A-2015-00109   
6       2016     3  A-2015-00108   
7       2016     3  A-2015-00107   
8       2016     3  A-2015-00082   
9       2016     3  A-2015-00106   
10      2016     3  A-2015-00100   
11      2016     3  A-2015-00098   
12      2016     3  A-2015-00085   
13      2016     3  A-2015-00084   
14      2016     3  A-2015-00063   
15      2016     3  A-2015-00061   
16      2016     3  A-2015-00058   
17      2016     3  A-2015-00057   
18      2016     3  A-2015-00052   
19      2016     3  A-2014-00183   
20      2016     3  A-2014-00182   
21      2016     3  A-2014-00161   
22      2016     3  A-2014-00160   
23      2016     7  A-2016-00071   
24      2017     7  A-2017-00063   
25      2016    10  A-2016-00121   
26      2016     4   A-2016-0002   
27      2016     4  A-2015-00112   
28      2016     4  A-2015-00103   
29      2016     4  A-2015-00101   
...      ...   ...           ...   
28264   2017     5       2017-01   
28265   2016    10       2016-07   
28266   2016    11       2016-20   
28267   2017     2       2016-22   
28268   2017     4       2016-21   
28269   2017     5       2016-12   
28270   2017     5       2016-14   
28271   2017     5       2016-15   
28272   2017    11       2016-05   
28273   2017    11       2016-13   
28274   2017     3  A-2016-00009   
28275   2016    10  A-2016-00003   
28276   2016    10  A-2016-00006   
28277   2016    10  A-2016-00007   
28278   2016    11  A-2016-00004   
28279   2016     8  A-2016-00002   
28280   2016     6  A-2016-00001   
28281   2017     2  A-2016-00011   
28282   2017     2  A-2016-00013   
28283   2017     3  A-2016-00014   
28284   2017     1  A-2016-00010   
28285   2017     4  A-2016-00012   
28286   2017     4  A-2016-00015   
28287   2017     5  A-2016-00016   
28288   2017     5  A-2016-00017   
28289   2017     7  A-2017-00002   
28290   2017     7  A-2017-00003   
28291   2017     7  A-2017-00004   
28292   2017    10  A-2017-00005   
28293   2017    12  A-2017-00009   

                                               resume_an  \
0      Provide a copy of the Transitional Briefing Bo...   
1      All communications between the AgriMarketing P...   
2      Copies  of any and all briefing materials prep...   
3      The amount of funding provided to National Far...   
4      All  records relating to a 2015-2016 outbreak ...   
5      Any communications, since January, 2015, betwe...   
6      Forensic audit report as described in Merx Sol...   
7      Pin-to-Pin messages between Pierre Corriveau a...   
8             Policy briefing book for the new minister    
9      Any communications on or instructions given to...   
10      A list of briefing materials, notes or memos ...   
11     All information about which company was funded...   
12     Copies of any briefing books or transition boo...   
13     A copy of the briefing book/binder/notes prepa...   
14     All briefing notes, memos, talking points, min...   
15     Provide access to last available draft or copy...   
16     Copies of any briefing books or transition boo...   
17     I would like to receive a copy of all briefing...   
18     Records created by AAFC Programs Branch (Farm ...   
19     Table 1: Projects Conducted With Neonicotinoid...   
20     Table 1: Projects Conducted With Neonicotinoid...   
21     Table 1: Projects Conducted With Neonicotinoid...   
22     Table 1: Projects Conducted With Neonicotinoid...   
23     The amount of funding provided to National Far...   
24     Copy of the briefing note (#226947) UPDATE ON ...   
25     Please provide copies of the tobacco farming d...   
26     Farm-level micro data solely for dairy, poultr...   
27     Any damage assessments, executive memos, brief...   
28     Provide information relatin

### Dans la prochaine section, je vais essayer d'isoler certaines colonnes pour tenter de trouver certaines informations.

In [7]:
info.n_de_requete.count()
#Ici, j'ai demandé au code de me donner le nombre de requête que compte
#Le graphique. J'aurais pu choisir n'importe quel case, mais j'ai ici su
#pposé que toutes les demandes avaient reçu un numéro d'identification

#J'ai voulu essayer de mettre du texte dans mon output. 
#(Ex: que ca affiche : (Nombres de demandes d'accès : XX) 
#Mais ça n'a pas fonctionné.

28278

Dans la prochaine cellule, je vais essayer de voir quel est l'organisme qui a recu le plus de demandes

In [8]:
info.proprietaire.value_counts()

Global Affairs Canada | Affaires mondiales Canada                                                                                                                1799
National Defence | Défense nationale                                                                                                                             1703
Employment and Social Development Canada | Emploi et Développement social Canada                                                                                 1538
Innovation, Science and Economic Development Canada | Innovation, Sciences et Développement économique Canada                                                    1395
Immigration, Refugees and Citizenship Canada | Immigration, Réfugiés et Citoyenneté Canada                                                                       1394
Privy Council Office | Bureau du Conseil privé                                                                                                                   1233
Natu

## On peut donc voir un apercu des organismes qui ont recu les plus de demande et le moins de demandes. 

### Le plus : Affaires mondiales Canada (1799 demandes)

### Le moins : Fondation Asie Pacifique du Canada

In [9]:
#En regardant le tableau, je vois qu'une des colones est une sorte
#D'acronyme du nom de l'organisme ou du ministère propriétaire des
#informations. Pour ma propre culture, je vais faire une deuxième fois
#la requête précédente pour voir si le nombre de requête des indentique.



In [10]:
info.owner_org.value_counts()

dfatd-maecd     1799
dnd-mdn         1703
esdc-edsc       1538
ic              1395
cic             1394
pco-bcp         1233
nrcan-rncan     1187
fin             1181
lac-bac         1010
hc-sc            975
cra-arc          910
cbsa-asfc        857
tc               830
jus              769
ec               706
pch              695
aandc-aadnc      676
ps-sp            675
tbs-sct          649
dfo-mpo          645
rcmp-grc         579
pwgsc-tpsgc      557
cfia-acia        554
pc               524
vac-acc          457
cnsc-ccsn        364
aafc-aac         343
ssc-spc          332
infc             330
csc-scc          318
                ... 
cnlopb            18
vfpa-apvf         17
ocl-cal           15
bdc               15
wdba-apwd         14
jccbi-pjcci       12
mai               12
cpc-cpp           10
oci-bec           10
nserc-crsng        9
fcac-acfc          9
cca-cac            9
prpa-appr          6
pspib-oirpsp       6
cnsopb             6
tpa-apt            6
cdic-sadc    

In [11]:
#Le résultat est probant, quand on regarde les deux listes,
#Le acronymes correspondent.

In [15]:
info.loc["annee","mois","resume_fr","proprietaire"]

KeyError: 'the label [annee] is not in the [index]'

In [16]:
#Pour une raison que j'ignore, cette fonction ne semble pas fonctionner

### Maintenant, je vais essyaer d'affichier seulement les données dont
### aucun docuement n'a été transmis. 



In [19]:
zero = info.disposition == "No records exist / Aucun documents existant"

In [20]:
info[zero]

annee  mois  n_de_requete  \
3     2016     7  A-2016-00076   
23    2016     7  A-2016-00071   
33    2016     7  A-2016-00072   
43    2016     7  A-2016-00077   
62    2016     7  A-2016-00082   
63    2016     7  A-2016-00078   
64    2016     7  A-2016-00075   
65    2016     7  A-2016-00074   
66    2016     7  A-2016-00073   
67    2016     7  A-2016-00070   
68    2016     7  A-2016-00068   
69    2016     7  A-2016-00065   
70    2016     7  A-2016-00063   
72    2016     7  A-2016-00060   
73    2016     7  A-2016-00052   
74    2016     7  A-2016-00043   
81    2016     7  A-2014-00196   
86    2016     7  A-2014-00189   
87    2016     7  A-2014-00176   
88    2016     7  A-2014-00175   
94    2016     8  A-2016-00094   
96    2016     8  A-2016-00092   
145   2017     1  A-2016-00158   

                                             resume_an  \
3    The amount of funding provided to National Far...   
23   The amount of funding provided to National Far...   
33   The amount of funding provided to National Far...   
43   The amount of funding provided to National Far...   
62   Any and all correspondence, including attachme...   
63   The amount of funding provided to the National...   
64   The amount of funding provided to National Far...   
65   The amount of funding provided to National Far...   
66   The amount of funding provided to National Far...   
67   Copies of all project proposals from National ...   
68   Copies of all project proposals from National ...   
69   Copies of all correspondence and documentation...   
70   Copies of all correspondence and documentation...   
72   Copies of  all communications between the Agri...   
73   Copies of all correspondence and documentation...   
74   Copies of all project proposals from National ...   
81   Table 1: Projects Conducted With Neonicotinoid...   
86   Table 1: Projects Conducted With Neonicotinoid...   
87   Table 1: Projects Conducted With Neonicotinoid...   
88   Table 1: Projects Conducted With Neonicotinoid...   
94   Documents, including any policy document(s), d...   
96   Any records, notes, memos or communications to...   
145  Please provide any correspondence, emails, mem...   

                                             resume_fr  \
3    Le montant du financement octroyé au Conseil n...   
23   Le montant du financement octroyé au Conseil n...   
33   Le montant du financement octroyé au Conseil n...   
43   Le montant du financement octroyé au Conseil n...   
62   Toute la correspondance, y compris les pièces ...   
63   Le montant du financement octroyé au Conseil n...   
64   Le montant du financement octroyé au Conseil n...   
65   Le montant du financement octroyé au Conseil n...   
66   Le montant du financement octroyé au Conseil n...   
67   Copies de tous les propositions de projets du ...   
68   Copies de toutes les propositions de projets d...   
69   Copies de l'ensemble de la correspondance et d...   
70   Copies de l'ensemble de la correspondance et d...   
72   Copies de toutes les communications entre les ...   
73   Copies de l'ensemble de la correspondence et d...   
74   Copies de toutes les propositions de projets d...   
81   Tableau 1 : Projets sur les néonicotinoïdes do...   
86   Tableau 1 : Projets sur les composés de néonic...   
87   Tableau 1 : Projets sur les néonicotinoïdes de...   
88   Tableau 1 : Projets sur les néonicotinoïdes de...   
94   Les documents, y compris les documents de poli...   
96   Les dossiers, notes, notes de service ou commu...   
145                                                NaN   

                                     disposition  pages owner_org  \
3    No records exist / Aucun documents existant      0  aafc-aac   
23   No records exist / Aucun documents existant      0  aafc-aac   
33   No records exist / Aucun documents existant      0  aafc-aac   
43   No records exist / Aucun documents existant      0  aafc-aac   
62   No records exist / Aucun documents existant      0  

In [ ]:
### Voila! J'aurais pu continuer longtemps mais voici la fin de ce devoir.